File: src/main/java/com/moovie/backend/domain/usecase/movies/GetLanguages.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.model.Language;
import com.moovie.backend.domain.repository.MoviesRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetLanguages {

    private final MoviesRepository moviesRepository;

    public GetLanguages(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public List<Language> invoke() {
        return moviesRepository.getLanguages();
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetLanguages.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.Language
import org.mobyle.domain.repository.MoviesRepository

class GetLanguages(private val repository: MoviesRepository) {
    operator fun invoke(): List<Language> = runBlocking {
        repository.getLanguages()
    }
}
